foodsaving/unsubscribe/api.py

```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth import get_user_model
+from django.utils.translation import ugettext as _
+from rest_framework import views, status
+from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+
+from foodsaving.conversations.models import ConversationMessage
+from foodsaving.unsubscribe.utils import parse_token, unsubscribe_from_all_conversations_in_group, \
+    unsubscribe_from_thread, unsubscribe_from_conversation
+
+
+class UnsubscribeView(views.APIView):
+    permission_classes = (AllowAny, )
+
+    @staticmethod
+    def post(request, token):
+        """
+        Receive unauthenticated but signed unsubscribe requests
+        These are the things people can click in emails regardless of whether they are logged in
+        """
+
+        # TODO: make it return a generic "invalid token" error in most cases
+
+        choice = request.data.get('choice', 'conversation')
```

Comment:
Why would it fall back to `conversation`?

Reply:
Because it's the least impactful thing you can do, and the first option in the list, and kind of the thing that the link in the email says it will do "Unsubscribe from this conversation".... but also could just make it mandatory to provide a choice.

Reply:
Frontend arrived!

Reply:
https://github.com/yunity/karrot-frontend/pull/1195